pkg/sds/server/metrics/metrics.go

```diff
@@ -0,0 +1,62 @@
+package metrics
```

Comment:
Do you think `metrics` should be nested into `server`? I'd also move it to `pkg/sds` directory:
- `/pkg/sds`
    - `ca`
        - `provider.go` 
    - `identity`
        - `provider.go` 
    - `metrics` 
    - `server`
        - `...`

And I'd rename to just `Metrics` because it's already under `sds` dir. Does it make sense?